1-Vanilla/scripts/todo_list.js

```diff
@@ -0,0 +1,130 @@
+  newItemContainer.id = "item-bar-" + id;
+
+  let newItemCheckBtn = createButton(id, "check", getCheckItemCallback(id));
+  newItemContainer.appendChild(newItemCheckBtn);
+
+  let newItemLabel = document.createElement("label");
+  newItemLabel.classList.add("item");
+  newItemLabel.id = "item-label-" + id;
+  newItemLabel.innerText = text;
+  newItemContainer.appendChild(newItemLabel);
+
+  let newItemEditBtn = createButton(id, "edit", getEditItemCallback(id));
+  newItemContainer.appendChild(newItemEditBtn);
+
+  let newItemRemoveBtn = createButton(id, "remove", getRemoveItemCallback(id));
+  newItemContainer.appendChild(newItemRemoveBtn);
+
+  return newItemContainer;
+}
+
+function addItem() {
+  let newItemInput = document.getElementById("new-item-name");
+
+  if (newItemInput.value.length > 0) {
+    let newItem = createTodoItem(nextId, newItemInput.value);
+
+    nextId++;
+
+    document.getElementById("main-container").appendChild(newItem);
+    document.getElementById("new-item-name").value = "";
+  }
+}
+
+function getRemoveItemCallback(id) {
+  return function () {
+    let elementToRemove = document.getElementById("item-bar-" + id);
+    document.getElementById("main-container").removeChild(elementToRemove);
+  };
+}
+
+function getCheckItemCallback(id) {
+  return function () {
+    let checkBtn = document.getElementById("check-btn-" + id);
+
+    if (checkBtn.classList.contains("checked")) {
+      console.log(checkBtn.classList);
+      checkBtn.classList.remove("checked");
+      console.log(checkBtn.classList);
+    } else {
+      checkBtn.classList.add("checked");
+    }
+  };
+}
+
+function getConfirmItemEditCallback(id) {
+  return function () {
+    let bar = document.getElementById("item-bar-" + id);
+    let input = document.getElementById("item-edit-" + id);
+    let confirmBtn = document.getElementById("confirm-btn-" + id);
+
+    let newTextLabel = document.createElement("label");
+    newTextLabel.classList.add("item");
+    newTextLabel.id = "item-label-" + id;
+    newTextLabel.innerText = input.value;
+    bar.replaceChild(newTextLabel, input);
+
+    let editBtn = createButton(id, "edit", getEditItemCallback(id));
+    bar.replaceChild(editBtn, confirmBtn);
+  };
+}
+
+function getEditItemCallback(id) {
+  return function () {
+    let bar = document.getElementById("item-bar-" + id);
+    let label = document.getElementById("item-label-" + id);
+    let editBtn = document.getElementById("edit-btn-" + id);
+    let oldText = label.innerText;
+
+    let newTextInput = document.createElement("input");
+    newTextInput.type = "text";
+    newTextInput.classList.add("edit");
+    newTextInput.id = "item-edit-" + id;
+    newTextInput.value = oldText;
+    bar.replaceChild(newTextInput, label);
+
+    let confirmBtn = createButton(
+      id,
+      "confirm",
+      getConfirmItemEditCallback(id)
+    );
+    bar.replaceChild(confirmBtn, editBtn);
+  };
+}
+
+window.addEventListener("load", (event) => {
```

Comment:
you could skip this callback if your `script` would be put on the bottom of the `index.html` file. it will ensure that the initial render was done